### PR TITLE
arch/risc-v: Attach exception handler in common place

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
* Align prototype of riscv_exception with xcpt_t 
* Add a common exception handler initializer to attach common exception
* Move epc adjustment to riscv_doirq
## Impact
Refactor only
## Testing
CI

### CODE REVIEW

File: arch/risc-v/src/common/riscv_common_irq.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#include <nuttx/irq.h>
+#include <sys/types.h>
+
+#include "riscv_internal.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void riscv_common_irq_attach(void)
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_common_irq.c:36]: can we move into riscv_initialize.c and call from up_initialize?

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_common_irq.c:36]: Yes, we can, but is it too late for exception ? There are many other device operation during irq_initialize (current ) and up_initialize

Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_common_irq.c:36]: Ok

File: arch/risc-v/src/common/riscv_common_irq.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  irq_attach(RISCV_IRQ_ECALLU, riscv_swint, NULL);
+#else
+  irq_attach(RISCV_IRQ_ECALLU, riscv_exception, NULL);
+#endif
+
+  irq_attach(RISCV_IRQ_ECALLS, riscv_exception, NULL);
+  irq_attach(RISCV_IRQ_ECALLH, riscv_exception, NULL);
+
+#ifndef CONFIG_ARCH_USE_S_MODE
+  irq_attach(RISCV_IRQ_ECALLM, riscv_swint, NULL);
```
Comment (no1wudi) [inline arch/risc-v/src/common/riscv_common_irq.c:59]: @pussuw Could you take a look ? Is it OK for S mode ?

Comment (pussuw) [reply arch/risc-v/src/common/riscv_common_irq.c:59]: It should be, ECALLS and ECALLM are handled by SBI.

File: arch/risc-v/src/bl602/Make.defs [hunk 3]
```diff
@@ -34,7 +34,7 @@ CMN_CSRCS += riscv_releasepending.c riscv_reprioritizertr.c
 CMN_CSRCS += riscv_releasestack.c riscv_stackframe.c riscv_schedulesigaction.c
 CMN_CSRCS += riscv_sigdeliver.c riscv_udelay.c riscv_unblocktask.c riscv_usestack.c
 CMN_CSRCS += riscv_idle.c riscv_tcbinfo.c riscv_getnewintctx.c riscv_doirq.c
-CMN_CSRCS += riscv_cpuindex.c
+CMN_CSRCS += riscv_cpuindex.c riscv_common_irq.c
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply arch/risc-v/src/bl602/Make.defs:37]: OK

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@no1wudi please fix the conflict.

**Comment by no1wudi** [comment]
> @no1wudi please fix the conflict.

OK

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**